From e2912161c7fc5c336baa963dc7442972a0e3c7ab Mon Sep 17 00:00:00 2001
From: Thierry Boureille <thierry.boureille@gmail.com>
Date: Thu, 28 Jul 2011 02:35:05 +0200
Subject: [PATCH 1/4] service : Fix put data on null pointer

when using ConnectService of Manager interface a segfault arises due
to a null pointer.

in __connman_service_create_and_connect:

	[...]
	service = lookup_by_identifier(name);

	if (service != NULL)
		goto done;

	network = create_hidden_wifi(device, ssid, mode, security);
	if (network != NULL) {
		connman_network_set_group(network, group);
		service->network_created = TRUE;
	}
	[...]
in this part "service" is null

Fixes BMC#21709
---
 src/service.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/service.c b/src/service.c
index b8f6aac..4ec52cb 100644
--- a/src/service.c
+++ b/src/service.c
@@ -4211,10 +4211,8 @@ int __connman_service_create_and_connect(DBusMessage *msg)
 		goto done;
 
 	network = create_hidden_wifi(device, ssid, mode, security);
-	if (network != NULL) {
+	if (network != NULL)
 		connman_network_set_group(network, group);
-		service->network_created = TRUE;
-	}
 
 	service = lookup_by_identifier(name);
 
@@ -4227,6 +4225,8 @@ done:
 		goto failed;
 	}
 
+	service->network_created = TRUE;
+
 	if (is_connected(service) == TRUE) {
 		err = -EISCONN;
 		goto failed;
-- 
1.7.6

